src/state.rs

```diff
@@ -336,14 +383,14 @@ impl ProxyState {
                         {
                             return None;
                         }
-                        Some((rank, ep))
+                        Some((rank, ep, wl))
                     })
                     .collect::<Vec<_>>();
-                let max = *ranks.iter().map(|(rank, _ep)| rank).max()?;
+                let max = *ranks.iter().map(|(rank, _ep, _wl)| rank).max()?;
```

Comment:
Probably a cute way to chain these `map()`s in one statement but I suspect it would be a cosmetic difference.